src/bin/stats.rs

```diff
@@ -19,25 +24,35 @@ pub struct FrameSummary {
   pub size: usize,
   pub input_frameno: u64,
   pub frame_type: FrameType,
-  /// PSNR for Y, U, and V planes
-  pub psnr: Option<(f64, f64, f64)>,
+  /// Contains metrics such as PSNR, SSIM, etc.
+  pub metrics: QualityMetrics,
   /// QP selected for the frame.
   pub qp: u8,
   /// Block-level encoding stats for the frame
   pub enc_stats: EncoderStats,
 }
 
-impl<T: Pixel> From<Packet<T>> for FrameSummary {
-  fn from(packet: Packet<T>) -> Self {
-    Self {
-      size: packet.data.len(),
-      input_frameno: packet.input_frameno,
-      frame_type: packet.frame_type,
-      psnr: packet.psnr,
-      qp: packet.qp,
-      enc_stats: packet.enc_stats,
-    }
-  }
+pub fn build_frame_summary<T: Pixel>(
+  packets: Packet<T>, metrics_config: VideoDetails,
+) -> Result<Option<FrameSummary>, CliError> {
+  let metrics_input_frame: &Frame<T> = packets.source.as_ref().unwrap();
+  let metrics_output_frame: &Frame<T> = packets.rec.as_ref().unwrap();
+  let metrics_cli = parse_cli().unwrap().metrics_enabled;
```

Comment:
The `parse_cli` function seems like it would be somewhat expensive to call on every frame. I think it would be better to pass the `MetricsEnabled` that we get from parsing the CLI args at the start of the encode down as an argument through `process_frame` into this function.

Reply:
In the latest revision it is already fixed and it is now an argument in `build_frame_summary`. Can you please see it again